### ISSUE TITLE
using x-ray with aws-sdk-v2 makes the binary files bigger than expected

### ISSUE THREAD

**Issue by kuadrosx** [aws/aws-xray-sdk-go#470]
Hello,

I was migrating from aws-sdk-v1(in July 31, 2024 it will be in maintance mode) to aws-sdk-v2 and I noticed that the binary files got twice its size.

Using https://pkg.go.dev/github.com/Zxilly/go-size-analyzer I find out that it was because both sdk version were imported:
```
| 10.62%  | github.com/aws/aws-sdk-go-v2                 | 3.8 MB | vendor    |
| 6.02%   | github.com/aws/aws-sdk-go                    | 2.1 MB | vendor    |
```

after finish the migration I continue seeing aws-sdk-go being added to the binary, so using https://github.com/KyleBanks/depth I learn that xray was importing both SDKs.

I reproduced the issue using the sample app defined in this repository and adding 3 more version:

https://github.com/kuadrosx/aws-xray-sdk-go/tree/reproduce-big-files-issue/sample-apps

these are the sizes of each version:
```
 15M	./http-server-sdk-v1-no-xray
 25M	./http-server-sdk-v1
 11M	./http-server-sdk-v2-no-xray
 34M	./http-server-sdk-v2
 86M	.
```

I think the expected size should be around 21M or less.

**Comment by wangzlei** [comment]
Yes, the xray go sdk still [using aws-sdk-v1 for sampling](https://github.com/aws/aws-xray-sdk-go/blob/master/strategy/sampling/proxy.go#L12-L16), it cause aws-sdk-v2 only customer's binary has to include aws-sdk-v1 dependencies. 

I create an issue that migrate aws sdk v1 in sampling module to v2 https://github.com/aws/aws-xray-sdk-go/issues/473

**Comment by jufemaiz** [comment]
Is there a timeline to removing entirely v1 given the timeline for v1 support being dropped?

**Comment by jj22ee** [comment]
`AWS SDK Go v1` has been removed in v2.0.0 of the X-Ray SDK for Go:
https://github.com/aws/aws-xray-sdk-go/blob/master/CHANGELOG.md#release-v200-2025-02-19
